File: Backend/post/post-java-springboot-maven-postgresql/src/main/java/com/projectbible/post/maven/postgresql/common/api/HealthController.java
package com.projectbible.post.maven.postgresql.common.api;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import java.util.Map;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Tag(name = "health")
@RestController
@RequestMapping("/api/v1/health")
public class HealthController {
    @Operation(summary = "Health check")
    @GetMapping
    public ApiResponse<Map<String, Object>> health() {
        return ApiResponse.success(Map.of(
            "status", "UP",
            "service", "post-java-springboot-maven-jpa-postgresql",
            "domain", "post"
        ));
    }
}

File: Backend/post/post-typescript-nestjs-npm-postgresql/src/common/health.controller.ts
import { Controller, Get } from "@nestjs/common";
import { ApiOperation, ApiTags } from "@nestjs/swagger";
import { ok } from "./api-response";
import { Public } from "./public.decorator";

@ApiTags("health")
@Controller("api/v1/health")
export class HealthController {
  @Public()
  @Get()
  @ApiOperation({ summary: "Health check" })
  health() {
    return ok({
      status: "UP",
      service: "post-typescript-nestjs-npm-postgresql",
      domain: "post",
    });
  }
}
